src/chef_authn.erl

```diff
@@ -115,8 +116,16 @@ process_key({'Certificate', _Der, _} = CertEntry) ->
     TbsCert = Cert#'Certificate'.tbsCertificate,
     Spki = TbsCert#'TBSCertificate'.subjectPublicKeyInfo,
     {0, KeyDer} = Spki#'SubjectPublicKeyInfo'.subjectPublicKey,
-    public_key:der_decode('RSAPublicKey', KeyDer).
-
+    public_key:der_decode('RSAPublicKey', KeyDer);
+process_key({'PrivateKeyInfo', _, _} = Entry) ->
+    KeyInfo = public_key:pem_entry_decode(Entry),
+    KeyAlgorithmInfo = KeyInfo#'PrivateKeyInfo'.privateKeyAlgorithm,
+    #'PrivateKeyInfo_privateKeyAlgorithm'{algorithm=Algorithm} = KeyAlgorithmInfo,
+    case Algorithm of
```

Comment:
The !@#$  is up with that record structure...
This might be a bit "better", given how interesting the record structure is... 
```erlang
case KeyAlgorithmInfo of 
    #'PrivateKeyInfo'{privateKeyAlgorithm = #'PrivateKeyInfo_privateKeyAlgorithm'{algorithm = ?'rsaEncryption'}, 
                      privateKey = PrivateKey} -> 
        public_key:der_decode('RSAPrivateKey', list_to_binary(PrivateKey));
    _  -> 
            % we'll need to handle the case when it's not rsaEncryption or else we'll crash here. 
        sadpanda
end.